src/cli/vector_enrich_doc.rs

```diff
@@ -164,11 +165,54 @@ fn vector_enrich_doc(
                     }
                 };
 
-                let doc_json = match rec.get_index(0).unwrap().1.as_record() {
-                    Ok(r) => r,
-                    Err(_) => {
-                        return Err(could_not_parse_input_error(span));
+                // Check if the input is from a doc get
+                let (doc_json, id) = if rec.contains("id")
+                    && rec.contains("content")
+                    && rec.contains("cas")
+                    && rec.contains("error")
+                    && rec.contains("cluster")
+                {
+                    // error is either Nothing which will result in an empty string or an error string
+                    // in either case the double unwrap here is safe
+                    let err = rec.get("error").unwrap().as_str().unwrap();
+                    if !err.is_empty() {
+                        return Err(generic_error(
+                            format!("error from doc get input: {}", err),
+                            None,
+                            None,
+                        ));
                     }
+
+                    (
+                        //Safe to unwrap as we have validated the presence of these cols in the record
+                        rec.get("content").unwrap().as_record()?,
+                        rec.get("id").unwrap().as_str()?.to_string(),
+                    )
+                } else {
+                    // Else piped input is from a query, which needs to contain 3 columns, one to be used as the ID, one holding the json doc and finally one with the cluster
+                    if rec.len() != 3 {
```

Comment:
I assume there is some implied behaviour here, but it's strange that the query example above only selects the ID and the contents (via *), but the 3rd field doesn't appear.  Where does the cluster come from?

Reply:
It's comes from cbsh, the return from a query has an added column with the cluster identifier against which the query was performed.

```
> query "SELECT meta().id, * FROM `travel-sample`.inventory.airline LIMIT 1"
╭───┬────────────┬──────────────────────────────┬─────────╮
│ # │     id     │           airline            │ cluster │
├───┼────────────┼──────────────────────────────┼─────────┤
│ 0 │ airline_10 │ ╭──────────┬───────────────╮ │ local   │
│   │            │ │ id       │ 10            │ │         │
│   │            │ │ type     │ airline       │ │         │
│   │            │ │ name     │ 40-Mile Air   │ │         │
│   │            │ │ iata     │ Q5            │ │         │
│   │            │ │ icao     │ MLA           │ │         │
│   │            │ │ callsign │ MILE-AIR      │ │         │
│   │            │ │ country  │ United States │ │         │
│   │            │ ╰──────────┴───────────────╯ │         │
╰───┴────────────┴──────────────────────────────┴─────────╯```